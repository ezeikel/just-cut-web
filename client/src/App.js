import React, { Component } from 'react';
import './App.css';

class App extends Component {
  state = { users: [] }

  async componentDidMount() {
    const res = await fetch('/users');
    const users = await res.json();
    this.setState({ users });
  }

  render() {
    return (
      <div className="App">
        <h1>Users</h1>
        {this.state.users.map(user =>
          <div key={user.id}>{user.username}</div>
        )}
      </div>
    );
  }
}

export default App;
