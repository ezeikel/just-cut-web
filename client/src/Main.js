import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Home from './Home';
import Shops from './Shops';
import Reviews from './Reviews';

const Main = () => (
    <main>
        <Switch>
            <Route exact path='/' component={Home} />
            <Route path='/shops' component={Shops} />
            <Route path='/reviews' component={Reviews} />
        </Switch>
    </main>
)

export default Main;
