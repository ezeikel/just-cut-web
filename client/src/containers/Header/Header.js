import React from 'react';
import { NavLink } from 'react-router-dom'

import './Header.css';

const Header = () => (
    <header>
        <nav>
            <ul>
                <li><NavLink to="/" exact>Home</NavLink></li>
                <li><NavLink to="/shops">Shops</NavLink></li>
                <li><NavLink to="/reviews">Reviews</NavLink></li>
            </ul>
            <ul>
                <li><NavLink to="/add-shop">Add Shop</NavLink></li>
            </ul>
        </nav>
    </header>
);

export default Header;
