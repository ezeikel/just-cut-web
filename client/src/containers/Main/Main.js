import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Home from '../Home/Home';
import Shops from '../../components/Shops/Shops';
import AddShop from '../../components/AddShop/AddShop';
import Shop from '../../components/Shops/Shop/Shop';
import Register from '../Register/Register';
import Login from '../Login/Login';

const Main = () => (
  <main>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/register" component={Register} />
      <Route path="/login" component={Login} />
      <Route path="/shops" component={Shops} />
      <Route path="/shop/:slug" component={Shop} />
      <Route path="/add-shop" component={AddShop} />
    </Switch>
  </main>
);

export default Main;
