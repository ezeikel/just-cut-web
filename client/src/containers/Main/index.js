import React from 'react';
import { Switch, Route } from 'react-router-dom';
import AddShop from '../../components/AddShop';
import Shops from '../Shops';
import Reviews from '../Reviews';

const Main = () => (
    <main>
        <Switch>
            <Route exact path='/' component={AddShop} />
            <Route path='/shops' component={Shops} />
            <Route path='/reviews' component={Reviews} />
        </Switch>
    </main>
)

export default Main;
