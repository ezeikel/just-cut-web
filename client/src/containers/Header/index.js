import React from 'react';
import { Link } from 'react-router-dom'

import './Header.css';

const Header = () => (
    <header>
        <nav>
            <ul>
                <li><Link to="/">Home</Link></li>
                <li><Link to="/shops">Shops</Link></li>
                <li><Link to="/reviews">Reviews</Link></li>
            </ul>
        </nav>
    </header>
)

export default Header;
