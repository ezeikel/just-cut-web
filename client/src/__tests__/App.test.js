import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../containers/App/App';
import Header from '../containers/Header/Header';
import Main from '../containers/Main/Main';

describe('App Component', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });

  it('should match snapshot', () => {
    expect(shallowToJson(shallow(<App />))).toMatchSnapshot();
  });

  it('should render Header component', () => {
    expect(shallow(<App />).find(Header).exists()).toBe(true);
  });

  it('should render Main component', () => {
    expect(shallow(<App />).find(Main).exists()).toBe(true);
  });
});
