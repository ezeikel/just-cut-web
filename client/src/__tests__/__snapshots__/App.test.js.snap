// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<styled.div>
  <Header />
  <Main />
</styled.div>
`;
